### PR TITLE
[SPARK-13896][SQL]Dataset.toJSON should return Dataset[String]

### PR DESCRIPTION
**Pull request by skonto**
## What changes were proposed in this pull request?

Change the return type of toJson in Dataset class
## How was this patch tested?

No additional unit test required.

### CODE REVIEW

**Comment by rxin** [comment]
Thanks - going to merge this.

File: sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala [hunk 1]
```diff
@@ -1987,9 +1987,9 @@ class Dataset[T] private[sql](
    * @group rdd
    * @since 1.3.0
    */
-  def toJSON: RDD[String] = {
+  def toJSON: Dataset[String] = {
```
Comment (rxin) [inline sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala]: actually we'd need to update the documentation above.

**Comment by rxin** [comment]
Actually I'm just going to merge this and fix the doc myself later. Thanks.

**Comment by skonto** [comment]
i missed the docs ok...my bad

**Pull request closed without merging**